Common/ML/include/ML/OrtInterface.h

```diff
@@ -113,7 +113,7 @@ class OrtModel
  private:
   // ORT variables -> need to be hidden as pImpl
   struct OrtVariables;
-  OrtVariables* mPImplOrt;
+  std::shared_ptr<OrtVariables> mPImplOrt = nullptr;
```

Comment:
Why do you use a shared_ptr and no unique_ptr?

Reply:
Because I can't assign a nullptr in the header file since OrtVariables is not known there. It will throw an error for invalid size of incomplete type.

Reply:
You have to declare the constructor and destructor in the header, but instanciate it in the cxx.
I.e. only `OrtInterface()` in the header, and
`OrtInterface::OrtInterface() = default;` in the cxx.
Then you can use unique_ptr.

Reply:
Is there any reason why we need a unique_ptr? Especially for the OrtAllocator (which gets created only for one environment but used in multiple), shared_ptr would probably be preferable. Unique_ptr would also require a deleter, which will make the code more cumbersome...